spec/amp-shadow-doc.md

```diff
@@ -96,6 +96,42 @@ Both `AMP.attachShadowDoc` and `AMP.attachShadowDocAsStream` return a `ShadowDoc
 - `shadowDoc.close()` - closes the AMP document and frees the resources.
 
 
+## Shadow DOM API and polyfills
+
+AMP Shadow Docs rely heavily on [Shadow DOM API](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Shadow_DOM). This is a powerful and elegant API, part of Web Components family. It allows natural isolation between major parts of the page and, as such, an ideal tool for PWAs and AMP Shadow Docs.
+
+Shadow DOM is currently implemented in Chrome and newer Safari, so it has a solid modern support and definitely not ubiquitous. AMP Shadow Docs API polyfills the necessary parts of Shadow DOM.
+
+However, not all advanced Shadow DOM features are polyfilled by AMP Shadow Docs. In particular, [shadow slots](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Slot) are not polyfilled. If you'd like to use slots and similar advanced features, please use one of the Shadow DOM polyfill, such as [WebComponents.js](https://github.com/webcomponents/webcomponentsjs). If you do, we recommend the following code structure (using Web Components polyfill as an example):
+
+HTML:
+```html
+<script async src="https://cdn.ampproject.org/shadow-v0.js"></script>
+<script async src="https://cdnjs.cloudflare.com/ajax/libs/webcomponentsjs/1.0.3/webcomponents-sd-ce.js"></script>
```

Comment:
How does Shadow AMP ensure that the web components polyfill is used? How does it detect it? Couldn't it load before the polyfill here?

Reply:
I helped build out the AMP Shadow with Polyfill solution. Currently, we would tap into the `'WebComponentsReady'` event that is broadcasted to the window.

If you are speaking of the use case where one would not like not use the webcomponents polyfill, but would still like to use the amp ShadowDOM polyfill, then they would simpley not include the `sdReadyPromise` snippet of code.

Is this what you meant?

Reply:
It would work mostly correctly if AMP APIs are used before polyfill is loaded. However, below I recommend waiting until polyfill is loaded. See the part with `const sdReadyPromise =`. It's the most reliable way to ensure that (a) things are loaded as much in parallel as possible, and (b) the app waits for right moment.

Reply:
@torch2424 The idea is that if another polyfill is used, the `sdReadyPromise` would still be used, but it'd instead be switched according to the other polyfill API.

Reply:
Ah yes that makes sense, thanks :)